src/System.Net.Http/src/System/Net/Http/HttpMethod.cs

```diff
@@ -106,7 +122,12 @@ public override bool Equals(object obj)
 
         public override int GetHashCode()
         {
-            return _method.ToUpperInvariant().GetHashCode();
+            if (_hashcode == 0)
+            {
+                _hashcode = _method.ToUpperInvariant().GetHashCode();
```

Comment:
If you were to go that route, better than doing 7 string comparisons, why not:

    if (StringAlreadyUpper(_method)) { return _method.GetHashCode(); }

Where `StringAlreadyUpper` is basically a `for` loop with calls to `char.IsUpper()`. Then you're iterating over the string once without allocating anything new. Surely faster than a `switch` on a string, no?